api/host_query_db.py

```diff
@@ -38,6 +40,8 @@ def get_host_list(
         query = _find_hosts_by_hostname_or_id(hostname_or_id)
     elif insights_id:
         query = _find_hosts_by_canonical_fact("insights_id", insights_id)
+    elif os_major:
+        query = _find_hosts_by_system_profile_fact("os_release", os_major, '((\d)\.*\d*)')
```

Comment:
If we changed the parameter type to integer, we should stringify it for the regex matching.

```suggestion
        query = _find_hosts_by_system_profile_fact("os_release", str(os_major), '((\d)\.*\d*)')
```